protocol/xmrmaker/claim.go

```diff
@@ -135,41 +193,93 @@ func claimRelayer(
 		return ethcommon.Hash{}, err
 	}
 
-	abi, err := abi.JSON(strings.NewReader(contracts.SwapFactoryMetaData.ABI))
+	calldata, err := getClaimTxCalldata(relayerCommission, contractSwap, secret)
 	if err != nil {
 		return ethcommon.Hash{}, err
 	}
 
-	feeValue, err := calculateRelayerCommission(contractSwap.Value, relayerCommission)
+	txHash, err := rc.SubmitTransaction(contractAddr, calldata)
 	if err != nil {
 		return ethcommon.Hash{}, err
 	}
 
-	calldata, err := abi.Pack("claimRelayer", *contractSwap, secret, feeValue)
+	err = waitAndCheck(ctx, ec, txHash)
 	if err != nil {
 		return ethcommon.Hash{}, err
 	}
 
-	txHash, err := rc.SubmitTransaction(contractAddr, calldata)
+	return txHash, nil
+}
+
+func waitAndCheck(ctx context.Context, ec *ethclient.Client, txHash ethcommon.Hash) error {
+	waitDuration := time.Second
+	maxNotFound := 5
+	start := time.Now()
+
+	var notFoundCount int
+	// wait for inclusion
+	for {
+		// sleep before the first check, b/c we want to give the tx some time to propagate
+		// into the node we're using
+		err := common.SleepWithContext(ctx, waitDuration)
+		if err != nil {
+			return err
+		}
+
+		_, isPending, err := ec.TransactionByHash(ctx, txHash)
+		if err != nil {
+			// allow up to 5 NotFound errors, in case there's some network problems
+			if errors.Is(err, ethereum.NotFound) && notFoundCount != maxNotFound {
```

Comment:
Safe programming practices recommend using `>=` in the 2nd half of the condition above.

Reply:
done